ProcessLib/TwoPhaseFlowWithPP/TwoPhaseFlowWithPPLocalAssembler-impl.h

```diff
@@ -122,27 +109,67 @@ void TwoPhaseFlowWithPPLocalAssembler<
+
+        MPL::VariableArray variables;
+
+        variables[static_cast<int>(MPL::Variable::phase_pressure)] = pn_int_pt;
+        variables[static_cast<int>(MPL::Variable::capillary_pressure)] =
+            pc_int_pt;
+        variables[static_cast<int>(MPL::Variable::temperature)] = temperature;
+
+        auto const rho_nonwet = gas_phase.property(MPL::PropertyType::density)
+                                    .template value<double>(variables, pos, t);
+
+        auto const rho_wet = liquid_phase.property(MPL::PropertyType::density)
+                                 .template value<double>(variables, pos, t);
+
+        auto const Sw = medium.property(MPL::PropertyType::saturation)
+                            .template value<double>(variables, pos, t);
 
         _saturation[ip] = Sw;
 
-        double dSw_dpc = _process_data.material->getSaturationDerivative(
-            material_id, t, pos, pn_int_pt, temperature, Sw);
+        auto const dSw_dpc =
+            medium.property(MPL::PropertyType::saturation)
+                .template dValue<double>(
+                    variables, MPL::Variable::capillary_pressure, pos, t);
 
-        double const porosity = _process_data.material->getPorosity(
-            material_id, t, pos, pn_int_pt, temperature, 0);
+        auto const porosity = medium.property(MPL::PropertyType::porosity)
+                                  .template value<double>(variables, pos, t);
 
-        // Assemble M matrix
-        // nonwetting
-        double const drhononwet_dpn =
-            _process_data.material->getGasDensityDerivative(pn_int_pt,
-                                                            temperature);
+        auto const drhononwet_dpn =
+            gas_phase.property(MPL::PropertyType::density)
+                .template dValue<double>(variables,
+                                         MPL::Variable::phase_pressure, pos, t);
+
+        auto const k_rel =
+            medium.property(MPL::PropertyType::relative_permeability)
+                .template value<MPL::Pair>(variables, pos, t);
+
+        auto const k_rel_wet = k_rel[0];
+        auto const k_rel_nonwet = k_rel[1];
+
+        auto const mu_nonwet = gas_phase.property(MPL::PropertyType::viscosity)
+                                   .template value<double>(variables, pos, t);
+
+        auto const lambda_nonwet = k_rel_nonwet / mu_nonwet;
+
+        auto const mu_wet = liquid_phase.property(MPL::PropertyType::viscosity)
+                                .template value<double>(variables, pos, t);
+
+        auto const lambda_wet = k_rel_wet / mu_wet;
+
+        auto const perm = medium.property(MPL::PropertyType::permeability)
+                              .template value<double>(variables, pos, t);
+
+        GlobalDimMatrixType permeability = GlobalDimMatrixType::Zero(
+            _element.getDimension(), _element.getDimension());
+
+        permeability.diagonal().setConstant(perm);
```

Comment:
i think we could call the function `MaterialPropertyLib::formEigenTensor` here
`auto const& K = MaterialPropertyLib::formEigenTensor<GlobalDim>(              medium.property(MaterialPropertyLib::PropertyType::permeability).value(vars, pos, t));`